src/ipcache.cc

```diff
@@ -543,8 +543,16 @@ ipcache_entry::updateTtl(const unsigned int rrTtl)
                                 Config.positiveDnsTtl); // largest value allowed
 
     const time_t rrExpires = squid_curtime + ttl;
-    if (rrExpires < expires)
+    const auto firstUpdate = addrs.size() <= 1;
+    if (firstUpdate) {
```

Comment:
Please avoid the code bloat. Just do the check without adding extra variables.
```suggestion
    if (addrs.size() <= 1) {
```

Reply:
Done at [b6f23c6](https://github.com/squid-cache/squid/pull/1380/commits/b6f23c6890b3b97f5baee0033fd57d70b94eccc9).